Adding record-tab-init event

ThoWagen opened the pull request:
Similar to the `results-init` event this event allows to listen for loaded record tabs.

ThoWagen committed:
Adding record-tab-init event

demiankatz reviewed (changes requested):
Thanks, @ThoWagen, this makes sense to me!

Note that record tab code is also used in embedded_record.js (see how that code also calls registerTabEvents()). Is it possible we also need to emit the event there to support the accordion/tabs modes?

I also wonder if the registerTabEvents() function should be refactored to event handlers once this event is in place. That would probably be too much change for a minor release, but if you'd like to add the event in 10.1 and then open a follow-up 11.0 PR to refactor further, that might be an approach (if you think refactoring is a good idea at all).

ThoWagen committed:
added event after all registerTabEvents calls

ThoWagen commented:
I added the emission of the event after every call to `registerTabEvents`. 

I can refactor this as you suggested. But since there are probably still some months until 10.1 I would wait with it. Should I create a Jira ticket and assign it to me so that it does not get forgotten?

demiankatz commented:
> I added the emission of the event after every call to `registerTabEvents`.

Thank you! One question: I see in one situation, the container being passed is the whole document. Is it possible we should do something more targeted there? Without knowing how this will be used, I can only guess, but thought it was worth asking the question.

> I can refactor this as you suggested. But since there are probably still some months until 10.1 I would wait with it. Should I create a Jira ticket and assign it to me so that it does not get forgotten?

Yes, that seems like a reasonable approach if you're happy with it!

ThoWagen commented:
I don't think that in that case a specific tab is being processed. We could use the complete ".record-tabs" element as container. But I am not sure if we can get any more specific here.

ThoWagen commented:
I created the ticket https://openlibraryfoundation.atlassian.net/browse/VUFIND-1700 but am not allowed to assign it to me.

demiankatz commented:
> I don't think that in that case a specific tab is being processed. We could use the complete ".record-tabs" element as container. But I am not sure if we can get any more specific here.

Yeah, that's what I was thinking as a possibility -- I don't know if it makes a significant difference, though!

demiankatz commented:
> I created the ticket https://openlibraryfoundation.atlassian.net/browse/VUFIND-1700 but am not allowed to assign it to me.

You hadn't been added to the jira-developers group yet. I've corrected that and assigned the issue to you. You should be able to self-assign in future if you wish. Thanks!

ThoWagen committed:
changed container for record-tab-init in recordDocReady

ThoWagen commented:
I changed the container.

> You hadn't been added to the jira-developers group yet. I've corrected that and assigned the issue to you. You should be able to self-assign in future if you wish. Thanks!

Great, thank you :)

demiankatz reviewed (approved):
Thanks, @ThoWagen, this looks good to me now!
